GoogleSignIn/Sources/GIDAuthentication.h

```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-#import "GoogleSignIn/Sources/Public/GoogleSignIn/GIDAuthentication.h"
```

Comment:
At what point do you intend to complete the merge of `GIDAuthentication` into `GIDGoogleUser`?  Currently, both objects share the same `authState` reference, which creates complexity.

Reply:
I'm not sure how the [single responsibility principle](https://en.wikipedia.org/wiki/Single-responsibility_principle) is an argument for keeping `GIDAuthentication` here.  However, the general goal of [encapsulation](https://en.wikipedia.org/wiki/Encapsulation_(computer_programming)) feels like a relevant consideration to me.  I don't think we want two classes that both read and mutate the same `OIDAuthState` object if we can avoid it.

Note that dropping `GIDAuthentication` and merging it into `GIDGoogleUser` was an explicit goal of our reviewed design for this refactoring of `GIDGoogleUser`.

Reply:
In our design our goal is to merge the public GIDAuthentication API into GIDGoogleUser API. We will accomplish that goal in this PR. And we keep GIDAuthentication class as private. I see most of your comments are about why not remove it. Let me try to explain it here.

Now the private GIDAuthentication class handles all authorization related methods. That is what I mean to follow single responsibility principle. Encapsulation, as you mentioned, is the main reason. We hide all the implementation details about handling EMM and talking to OIDAuthorizationService to refresh OIDAuthState. This will help us use fake to replace OCMock in the next project. In the next project we will rename GIDAuthentication to GIDAuthorizationService class and provide a fake. Please see doc Refactor GIDGoogleUser to improve tests.
https://docs.google.com/document/d/12lFJBakatD9KdKrBph3B0mw8Zw2KnVrkkwil6fFhB14/edit#